src/Mavi.jl

```diff
@@ -52,9 +52,31 @@ struct System{T, C1<:SpaceCfg, C2<:DynamicCfg}
     num_p::Int
 end
 
+"""
+Checks if all particles are inside the given configuration space. If not, throws an error.
+"""
+function check_inside(state::State{T}, space_cfg::RectangleCfg) where {T}
```

Comment:
Eu acredito que parametrizar o tipo aqui não interfere em nada (o desempenho da função continua o mesmo). Só faz sentido parametrizar se tu vai utilizar o parâmetro, e como T não é acessado em nenhum momento, ele é inútil aqui.

Eu fazia isso no meu código antes, mas depois eu parei pra pensar e vi que só estava complicando a minha vida por nada. O motivo de o desempenho não mudar (mas o ideal seria testar na prática pra ver se estou certo) é o seguinte:
- Quando a função `check_inside(state::State, space_cfg::RectangleCfg)` é chamada com o tipo de `state`, digamos,  sendo `State{Float64}`, a Julia cria um método especializado dessa função com a assinatura `check_inside(state::State{Float64}, space_cfg::RectangleCfg)`, então nas próximas chamadas esse método vai ser executado. Parametrizar a função aqui apenas nos dá acesso ao tipo que está dentro de `State`, mas se nós não usamos ele, a parametrização é inútil.

Reply:
Justo, faz sentido